### PR TITLE
initial test plan for cl_khr_unified_svm

### PR DESCRIPTION
**Pull request by bashbaug**
Please note, this PR targets a cl_khr_unified_svm branch, not the main branch, as we discussed.

It has an initial proposed test plan, for review.

see: #2149

### CODE REVIEW

**Comment by bashbaug** [comment]
Discussed in the January 7th teleconference.  We're going to keep tracking the unified SVM test plan using this PR, at least for now.  We will use PR suggestions to propose changes or additions, or to tick off items on the task list, as various tests are merged.

File: cl_khr_unified_svm-test-plan.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+* [ ] `clSetKernelArgSVMPointer`
+    * Generally do not need a targeted test, because this API will be exercised by any tests using SVM pointers in kernels.
+    * [ ] Ensure at least one test for each SVM type passes a pointer offset from the base pointer and accesses it in a kernel.
+    * [ ] TODO: Pass a bogus pointer, execute the kernel, and verify that no error occurs as long as the pointer is not dereferenced?
+* [ ] `clEnqueueSVMFree` (see: `test_enqueue_api.cpp`)
+    * [ ] Include an event on the command and verify that the event type is `CL_COMMAND_SVM_FREE`.
+    * [ ] Allocate memory for each type and verify that it can be freed asynchronously.
+* [ ] `clEnqueueSVMMemcpy` (see: `test_enqueue_api.cpp`)
+    * [ ] Include an event on the command and verify that the event type is `CL_COMMAND_SVM_MEMCPY`.
+    * [ ] Test all combinations of SVM pointer and host pointer sources and destinations.
```
[the remark this answers is not in the document]
Comment (JohnKesapidesARM) [reply cl_khr_unified_svm-test-plan.md:172]: Small status update: Tests are in internal review atm.

[the remark this answers is not in the document]
Comment (JohnKesapidesARM) [reply cl_khr_unified_svm-test-plan.md]: @bashbaug Internal review is complete so I can share the test code now
Because it took so long, I now have tests for this whole section (Testing Existing SVM APIs) that I can share. What is the best way to share the tests?

[the remark this answers is not in the document]
Comment (bashbaug) [reply cl_khr_unified_svm-test-plan.md]: Awesome!  Can you make a PR to the cl_khr_unified_svm branch and we can go from there?

[the remark this answers is not in the document]
Comment (bashbaug) [reply cl_khr_unified_svm-test-plan.md]: Semi-related: I started thinking about writing some "corner case" tests yesterday evening, to test some of the weirder cases currently marked TODO, but I don't want to duplicate work.  Does your testing include these items?  Thanks!

[the remark this answers is not in the document]
Comment (JohnKesapidesARM) [reply cl_khr_unified_svm-test-plan.md]: I skipped everything marked with a TODO, so should be fine :)

[the remark this answers is not in the document]
Comment (bashbaug) [reply cl_khr_unified_svm-test-plan.md]: Great.  Here are the "corner case" tests:  https://github.com/KhronosGroup/OpenCL-CTS/pull/2436

File: cl_khr_unified_svm-test-plan.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+https://github.com/KhronosGroup/OpenCL-CTS/tree/main/test_conformance/SVM
+
+Some prior tests for the Intel USM extension [`cl_intel_unified_shared_memory`](https://registry.khronos.org/OpenCL/extensions/intel/cl_intel_unified_shared_memory.html) may be found here:
+
+https://github.com/intel/compute-samples/tree/master/compute_samples/tests/test_cl_unified_shared_memory
+
+## Test Plan
+
+### Consistency Check
```
Comment (bashbaug) [inline cl_khr_unified_svm-test-plan.md:19]: We don't currently have any consistency checks for unified SVM capabilities and OpenCL 2.0 SVM capabilities ([CL_DEVICE_SVM_CAPABILITIES](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_DEVICE_SVM_CAPABILITIES)).  Should we add this?  If so, do we want consistency checks in both directions, or in just one?

1. Old -> New: If a device exposes the OpenCL 2.0 SVM capabilities and supports unified SVM, then the device should support the same functionality via unified SVM.
2. New -> Old: If a device exposes functionality via unified SVM, then the device should support the same functionality via OpenCL 2.0 SVM capabilities, also.

Regarding (1), an implementation could conceivably / theoretically want to expose OpenCL 2.0 SVM capabilities for backwards compatibility, but not support the same capabilities for newer code (that may only be checking for unified SVM capabilities).  Is this a scenario we would want to support?

I cannot think of a reason an implementation would not want to do (2) off the top of my head.

Comment (bashbaug) [inline cl_khr_unified_svm-test-plan.md:19]: With the addition of the concurrent access atom size query, we should add a few more consistency checks:

* For each set of SVM capabilities, if the capabilities includes the concurrent access capability, then the concurrent access atom size must be non-zero.  Otherwise, when the capabilities does not include the concurrent access capability, the concurrent access atom size must be zero.
